eip_operator/src/controller/eip.rs

```diff
@@ -80,13 +93,207 @@ impl k8s_controller::Context for Context {
                     .public_ip
                     .as_ref()
                     .ok_or(Error::MissingPublicIp)?;
-                (allocation_id.to_owned(), public_ip.to_owned())
+                status.allocation_id = Some(allocation_id.clone());
+                status.public_ip_address = Some(public_ip.clone());
             }
             _ => {
                 return Err(Error::MultipleEipsTaggedForPod);
             }
         };
-        crate::eip::set_status_created(&eip_api, name, &allocation_id, &public_ip).await?;
+
+        // get potential attachments
+        let mut matched_pods: Vec<Pod> = vec![];
+        let mut matched_nodes: Vec<Node> = vec![];
+
+        match eip.spec.selector {
+            EipSelector::Pod { ref pod_name } => {
+                matched_pods = pod_api
+                    .list(
+                        &ListParams{field_selector: Some(format!("metadata.name={}", pod_name)), ..Default::default() }
+                    )
+                    .await?
+                    .into_iter()
+                    .filter(|pod| pod.metadata.deletion_timestamp.is_none())
+                    .collect::<Vec<Pod>>();
+                matched_pods.sort_unstable_by_key(|s| s.name_unchecked());
+            }
+            EipSelector::Node { ref selector } => {
+                let label_selector = selector
+                    .iter()
+                    .map(|(k, v)| format!("{}={}", k, v))
+                    .collect::<Vec<String>>()
+                    .join(",");
+                matched_nodes = node_api
+                    .list(
+                        &ListParams{ label_selector: Some(label_selector), ..Default::default()}
+                    )
+                    .await?
+                    .into_iter()
+                    .filter(|node| node.metadata().deletion_timestamp.is_none())
+                    .collect::<Vec<Node>>();
+                matched_nodes.sort_unstable_by_key(|s| s.name_unchecked());
+            }
+        }
+
+        // Check to make sure our resource still matches
+        // incase our selectors have updated, or the nodes/pods we have
+        // claimed have changed
+        // let correctly_associated =
+        let mut disassociate = false;
```

Comment:
aren't we always setting `disassociate` to `status.claim.is_none()` here? do we need the separate variable?

Reply:
no not quite... if we don't have a claim, but we are associated with a pod that we match to we do not want to disassociate we want to update the claim.